src/figuresClasses.ts

```diff
@@ -1,19 +1,72 @@
 export interface Figure {
-
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Sides of a triangle must be greater than 0.');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('Invalid triangle sides.');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
 
+    return Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+  }
 }
 
-export class Circle {
+export class Circle implements Figure {
+  shape: 'circle' = 'circle';
 
+  constructor(public color: 'red' | 'green' | 'blue', public radius: number) {
+    if (radius <= 0) {
+      throw new Error('Radius of a circle must be greater than 0.');
+    }
+  }
+
+  getArea(): number {
+    const DECIMAL_PLACES = 2;
+
+    return Math.floor((Math.PI * this.radius * this.radius)
+     * (10 ** DECIMAL_PLACES)) / (10 ** DECIMAL_PLACES);
+  }
 }
 
-export class Rectangle {
+export class Rectangle implements Figure {
+  shape: 'rectangle' = 'rectangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
```

Comment:
it would be prefect if you'd access shape types through figure: `Figure['typeName']`